docs/current/guides/592101-custom-callbacks.md

```diff
@@ -0,0 +1,109 @@
+---
+slug: /592101/custom-callbacks
+displayed_sidebar: "current"
+category: "guides"
+tags: ["nodejs", "go", "python", ]
+authors: ["Vikram Vaswani"]
+date: "2023-08-18"
+---
+
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+# Use Custom Callbacks in a Dagger Pipeline
+
+## Introduction
+
+All Dagger SDKs support adding callbacks to the pipeline invocation chain. Using a callback enables greater code reusability and modularity, and also avoids "breaking the chain" when constructing a Dagger pipeline.
+
+This guide explains the basics of creating and using custom callbacks in a Dagger pipeline. You will learn how to:
+
+- Create a custom callback
+- Chain the callback into a Dagger pipeline
+
+## Requirements
+
+This tutorial assumes that:
+
+- You have a Go, Python or Node.js development environment. If not, install [Go](https://go.dev/doc/install), [Python](https://www.python.org/downloads/) or [Node.js](https://nodejs.org/en/download/).
+- You have a Dagger SDK installed for one of the above languages. If not, follow the installation instructions for the Dagger [Go](../sdk/go/371491-install.md), [Python](../sdk/python/866944-install.md) or [Node.js](../sdk/nodejs/835948-install.md) SDK.
+- You have Docker installed and running on the host system. If not, [install Docker](https://docs.docker.com/engine/install/).
+
+## Example
+
+All Dagger SDKs support adding a callback via the `With()` API method. The callback must return a function that receives a `Container` from the chain, and returns a `Container` back to it.
```

Comment:
Technically it's not an "API" method. It's not in GraphQL, it's added by each SDK in codegen.

Reply:
If anyone's curious, the method is added to every type that has at least one field that returns the same type (i.e., allows chaining).